### DISCUSSION TITLE
rtk-query for  svelte `toolkit/rtk-query/svelte`

### DISCUSSION THREAD

**Discussion by FaberVitale** [reduxjs/redux-toolkit#2027] [Ideas]
I've been thinking about a rtk-query for svelte, basically like `packages/toolkit/rkt-query/react` but for svelte.

I've taken a look at the svelte example https://redux-toolkit.js.org/rtk-query/usage/examples#svelte and is, imo, a bit too cumbersome compared to the custom react hooks.

I think that's doable, see https://github.com/SvelteStack/svelte-query, but is obviously not a small task.

What do you think? Feedback? Ideas?

**Comment by markerikson** [comment]
There's already https://github.com/SaulMoro/ngrx-rtk-query (and https://github.com/wobsoriano/vue-rtk-query ), so yeah, there's precedent of wrapping RTKQ for other frameworks.

**Reply by FaberVitale** [reply]
If possible I'd like to include rtk-query/svelte in `reduxjs/toolkit`:

we could develop initially as a  `RTK Incubator` project.

**Reply by markerikson** [reply]
I'd be hesitant to include it directly in RTK, tbh.  I'm not sure there's enough people who are using Svelte _and_ RTK to justify it.

**Reply by phryneas** [reply]
We also have the @rtk-query prefix used for [@rtk-query/codegen-openapi](https://www.npmjs.com/package/@rtk-query/codegen-openapi) and [@rtk-query/graphql-request-base-query](https://www.npmjs.com/package/@rtk-query/graphql-request-base-query)

**Comment by multiplehats** [comment]
I'd love to see this. I just discovered how well Redux Toolkit works with Svelte with not that much boilerplate.

But Svelte-like hooks would be absolutely brilliant.
